tests/exporters/test_s5.py

```diff
@@ -0,0 +1,299 @@
+            max_month=max_month,
+        )
+
+        # CHECK default arguments
+        assert (
+            processed_selection_request["originating_centre"] == "ecmwf"
+        ), "\
+        Expected originating_centre to be: {'ecmwf'}. Got:\
+        {processed_selection_request['originating_centre']}"
+
+        assert (
+            processed_selection_request["system"] == "5"
+        ), f"\
+        Expected 'system' to be '5'. Got:\
+        {processed_selection_request['system']}"
+
+        # CHECK time arguments
+        assert processed_selection_request["year"] == [
+            "2017",
+            "2018",
+        ], f"\
+        Expected 'year' to be ['2017','2018']. Got:\
+        {processed_selection_request['year']}"
+
+        exp_months = [
+            "{:02d}".format(month) for month in range(min_month, max_month + 1)
+        ]
+        assert (
+            processed_selection_request["month"] == exp_months
+        ), f"\
+        Expected 'month' to be {exp_months}. Got:\
+        {processed_selection_request['month']}"
+
+    def test_expected_filepath(self, tmp_path):
+        data_dir = Path(f"{tmp_path}/data")
+        granularity = "monthly"
+        pressure_level = False
+
+        s5 = S5Exporter(
+            data_folder=data_dir, granularity=granularity, pressure_level=pressure_level
+        )
+
+        expected_filepath = "data/raw/seasonal-monthly-single-levels/\
+        total_precipitation/2017_2018/M01_12.grib"
+
+        variable = "total_precipitation"
+        max_leadtime = 5
+        min_year = 2017
+        max_year = 2018
+        min_month = 1
+        max_month = 12
+
+        processed_selection_request = s5.create_selection_request(
+            variable=variable,
+            max_leadtime=max_leadtime,
+            min_year=min_year,
+            max_year=max_year,
+            min_month=min_month,
+            max_month=max_month,
+        )
+
+        assert (
+            expected_filepath == filepath.as_posix()
+        ), f"\
+            Expected: {expected_filepath}. Got: {filepath.as_posix()}"
+
+    def test_dataset_created_properly(self, tmp_path):
+        data_dir = Path(f"{tmp_path}/data")
+
+        expected_datasets = [
+            "seasonal-monthly-single-levels",
+            "seasonal-monthly-pressure-levels",
+            "seasonal-original-single-levels",
+            "seasonal-original-pressure-levels",
+        ]
+        pressure_levels = [False, True, False, True]
+        granularities = ["monthly", "monthly", "hourly", "hourly"]
+
+        for ix, expected in enumerate(expected_datasets):
+            s5 = S5Exporter(
+                data_folder=data_dir,
+                granularity=granularities[ix],
+                pressure_level=pressure_levels[ix],
+            )
+
+            assert (
+                s5.dataset == expected
+            ), f"\
+            Expected: {expected}. Got: {s5.dataset}"
+
+    @patch("cdsapi.Client")
```

Comment:
@gabrieltseng could you possibly help with this pytest patching malarkey? unfortunatley it's super tricky because my pytest is broken - but i ran these tests as simple assert statements in my ipython console. i really need to get stuck into an IDE